### ISSUE TITLE
Bitmap constructor with IntPtr fails unless stride & 3 == 0

### ISSUE THREAD

**Issue by ispysoftware** [dotnet/runtime#57125]
Using the bitmap constructor and passing in a pointer to raw data will fail unless the stride of the image & 3 is equal to 0.

for example calling:
var bmp = new Bitmap(1310, 688, 3930, PixelFormat.Format24bppRgb, ptrToUnmanagedData)
gives a "parameter is not valid" error as 3930 & 3 !=0

whereas calling
var bmp = new Bitmap(2316, 688, 6948, PixelFormat.Format24bppRgb, ptrToUnmanagedData)
works as 6948 & 3 == 0

This is a major issue as it means you can't work with shared pixel data from other libraries without doing a full memory copy back and forth for every frame when stride & 3 != 0

### Configuration

Using System.Drawing.Common v5.0.2 but tried latest running on .Net Standard x64

### Regression?
This has never worked

**Comment by koszeggy** [comment]
> gives a "parameter is not valid" error as 3930 & 3 !=0

The problem is not that 3930 & 3 !=0 but that 3930 % 4 !=0. GDI+ bitmaps must have a stride that is a multiple of 4. So `new Bitmap(1310, 688, 3932, PixelFormat.Format24bppRgb, ptrToUnmanagedData)` actually works, but this requires that all rows have a 2 bytes padding in the raw data so every row starts on an address that is dividable by 4.

This is not a .NET feature so changing this would require to change the underlying GDI+ functionality.

If your raw data contains continuous pixel stream without padding, the best you can do is to fill up a new `Bitmap` pixel by pixel. You can use [this](https://github.com/koszeggy/KGySoft.Drawing#fast-bitmap-manipulation) library (disclaimer: written by me) to do it in a performant, parallelizable way.

**Comment by ispysoftware** [comment]
ok understood thanks for the info, i'll see if i can wrangle the other library to force a padding. I'm guessing this is heading to won't fix?

**Comment by safern** [comment]
> ok understood thanks for the info, i'll see if i can wrangle the other library to force a padding. I'm guessing this is heading to won't fix?

Unfortunately yes, the reason for that being the answer is that System.Drawing.Common is only a thin wrapper around `GDI+` and we only support what `GDI+` does. 

Thanks for raising the issue and I hope @koszeggy information helps you get unblocked, if not feel free to continue the discussion and ask more questions.
